nixos/modules/services/monitoring/staytus.nix

```diff
@@ -0,0 +1,370 @@
+app_name: staytus
+processes:
+  web:
+    restart_mode: usr2
+  worker:
+    restart_mode: start-term
+log_path: /run/staytus/log/staytus.log
+log_root: /run/staytus/log
+pid_root: /run/staytus/pids
+env:
+${concatStrings (mapAttrsToList (name: value: "  ${name}: ${value}\n") staytusEnv)}
+'';
+
+ procfile = ''
+web: ${staytus-rake}/bin/staytus-puma
+worker: ${staytus-rake}/bin/staytus-jobs
+  '';
+
+  unpackTheme = unpack "theme";
+  unpack = id: (name: source:
+    pkgs.stdenv.mkDerivation {
+      name = "staytus-${id}-${name}";
+      buildInputs = [ pkgs.unzip ];
+      buildCommand = ''
+        mkdir -p $out
+        cd $out
+        unpackFile ${source}
+      '';
+  });
+
+  staytusEnv = {
+    RAILS_ENV = "production";
+    RACK_ENV = "production";
+    SECRET_KEY_BASE = cfg.secretKeyBase;
+    RAILS_LOG = "/run/staytus/log";
+    HOME = "/run/staytus";
+    STAYTUS_LANG = "en";
+    STAYTUS_THEME = "${cfg.theme}";
+    STAYTUS_DEMO = "0";
+    STAYTUS_SMTP_HOSTNAME = "${cfg.smtpHost}";
+    STAYTUS_SMTP_PORT = "${cfg.smtpPort}";
+    STAYTUS_SMTP_USERNAME = "${cfg.smtpUsername}";
+    STAYTUS_SMTP_PASSWORD = "${cfg.smtpPassword}";
+    GEM_HOME = "${package}/${ruby.gemPath}";
+    PUMA_DEBUG = if cfg.pumaDebug then "1" else "0";
+    RAILS_LOG_TO_STDOUT = "1";
+    APP_ROOT = "/run/staytus";
+  };
+
+  staytus-rake = pkgs.stdenv.mkDerivation rec {
+    name = "staytus-rake";
+    buildInputs = [ package.env pkgs.makeWrapper ];
+    phases = "installPhase fixupPhase";
+    installPhase = ''
+      mkdir -p $out/bin
+      makeWrapper ${package.env}/bin/bundle $out/bin/staytus-bundle \
+          ${concatStrings (mapAttrsToList (name: value: "--set ${name} '${value}' ") staytusEnv)} \
+          --set PATH '${lib.makeBinPath (with pkgs; [ nodejs ])}:$PATH' \
+          --set RAKEOPT '-f /run/staytus/Rakefile' \
+          --run 'env' \
+          --run 'cd /run/staytus'
+      makeWrapper $out/bin/staytus-bundle $out/bin/staytus-rake \
+          --add-flags "exec rake"
+      makeWrapper $out/bin/staytus-rake $out/bin/staytus-jobs \
+          --add-flags "jobs:work"
+      makeWrapper $out/bin/staytus-bundle $out/bin/staytus-puma \
+          --add-flags "exec puma" \
+          --add-flags "--environment production" \
+          --add-flags "-C /run/staytus/config/puma.rb" \
+          --add-flags "--redirect-stdout ${cfg.stateDir}/log/puma.log" \
+          --add-flags "--redirect-stderr ${cfg.stateDir}/log/puma.error.log" \
+          --add-flags "--redirect-append" \
+          --add-flags "--pidfile /run/staytus/pids/puma.pid"
+     '';
+  };
+
+in {
+
+  options = {
+    services.staytus = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Enable the staytus service.
+        '';
+      };
+
+      stateDir = mkOption {
+        type = types.str;
+        default = "/var/staytus";
+        description = "The state directory, logs and plugins are stored here";
```

Comment:
Is there an important use-case for making this configurable? If we advertise this as an API it is harder to change the location in future without breaking compatibility. What does upstream proposes here as an directory? I would use `/var/lib/staytus` otherwise.

Reply:
I think that the users should be able to decide where their logs go. I'll set the default to `/var/lib/staytus`.

Reply:
Done